### ISSUE TITLE
Newer Games (Megami Meguri) Requiring SecureData Value for Proper Restore

### ISSUE THREAD

**Issue by Jaws12** [J-D-K/JKSM#42]
With the assistance of users in #3dshacks on IRC, we have verified that the recently released Japanese eShop only game, Megami Meguri, will consider a save corrupted if the SecureData is blank or doesn't match with values stored in the save. It would seem for some newer games simply blanking out the SecureData value upon save restore like JKSM currently does may no longer be adequate.

**Comment by J-D-K** [comment]
Yes, I'm aware. I added a quick secure value export function, and it *appears* that 'wm.dat' is encrypted. I can't find the secure value in the file and it's not in any of the extdata either. If I could find it, it wouldn't be any trouble to just request it from the system and replace it, skipping the deletion completely.

**Comment by Jaws12** [comment]
Thanks for the reply! Good to know you are aware of the issue. Thinks there's any way we could figure out the encryption? I hope my save backup (without secureData) may eventually be recoverable, but at least hopefully my sacrifice will have not been in vain to bring this issue to light.

**Comment by yellows8** [comment]
Do you have a sample savefile?(Already had the disasm for this game's v2.0 update open)

**Comment by Jaws12** [comment]
I only have v1 save files (blank ones with user/ext/securedata and my currently unusable user/ext only) as I haven't been able to install the 2.0 update as of yet. Would those be useful? If so, what would be the best way to send them to you?

**Comment by yellows8** [comment]
Github attachment would be fine.

**Comment by Jaws12** [comment]
You will find an example blank/default save and my original progressed save attached to this message. Thanks for giving these a look!

[DefaultBlankSave-MegamiMeguri.zip](https://github.com/J-D-K/JKSM/files/778375/DefaultBlankSave-MegamiMeguri.zip)
[OriginalWorkingSave-MegamiMeguri.zip](https://github.com/J-D-K/JKSM/files/778376/OriginalWorkingSave-MegamiMeguri.zip)

**Comment by yellows8** [comment]
From L_505078():

`key = CRC32(<output from https://www.3dbrew.org/wiki/Cfg:GenHashConsoleUnique>);//CRC is not inverted at the end.`
`crypto_init(state);`
`cypto_setkey(state, key);`
`for(i=0; i<rounds; i++)crypto_update(state);`
`for(pos; pos<bufsize; pos++){crypto_update(state); buf8[pos] ^= crypto_update(state);}`

bufsize is slightly smaller than the actual filesize.

Don't know what algo is used for crypto and not interested in reimplementing it either.

**Comment by MelonGx** [comment]
@Jaws12 @J-D-K @yellows8 
Does this issue still active?
Just confirmed that this game's save was linked to your CTRNAND.
If you save it once, it will modify such CTRNAND files like the following image shows.
If all of those CTRNAND files are mismatch, the game will report a savedata corruption.
[https://i.imgur.com/XEljb1F.png](url)
